### PR TITLE
tests: drivers: Add nrfx integration test application

### PR DESCRIPTION
**Pull request by anangl**
The purpose of this test is to check if the nrfx adaptation layer
provided in nrfx_glue.h is correct and all nrfx drivers available
for particular nRF SoCs can be successfully compiled, and to perform
these checks as a part of CI testing.

### CODE REVIEW

**Comment by carlescufi** [comment]
Note: This is an automated message.
@anangl this Pull Request has not been updated for more than a month, please close it if not applicable anymore.
@carlescufi  please close it if the author is not available

**Comment by anangl** [comment]
Rebased on master.

**Comment by anangl** [comment]
Updated the application and switched it back to be a test instead of a sample.

**Comment by anangl** [comment]
Rebased on master again, to get revision cb9e75ebfea8a8312678b2359c6b41b051a95177 of the hal_nordic module.

**Comment by anangl** [comment]
Rebased on master.

**Comment by anangl** [comment]
Rebased on master.

**Comment by anangl** [comment]
Renamed `BT_LL_NRFXLIB_DEFAULT` to `BT_LL_SOFTDEVICE_DEFAULT`.

**Comment by anangl** [comment]
Enabled `NRFX_PRS_BOX_*` options.

File: tests/drivers/nrfx_integration_test/prj.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+CONFIG_ASSERT=y
+CONFIG_BT=y
+CONFIG_BT_LL_SOFTDEVICE_DEFAULT=y
```
Comment (PerMac) [inline tests/drivers/nrfx_integration_test/prj.conf:9]: I think you might need adding `CONFIG_ZTEST=y` (https://docs.zephyrproject.org/latest/guides/test/ztest.html#quick-start-integration-testing)

**Comment by PerMac** [comment]
What do you think about adopting this to sanitycheck standards (https://docs.zephyrproject.org/latest/guides/test/ztest.html#quick-start-integration-testing) i.e. using ztest, having the name starting with `test_` etc? I am worrying that this test won't be properly recognized by the sanitycheck, but I haven't check it yet. We can discuss it furtherly in priv.

**Comment by anangl** [comment]
Rebased on master.

**Comment by anangl** [comment]
> What do you think about adopting this to sanitycheck standards (https://docs.zephyrproject.org/latest/guides/test/ztest.html#quick-start-integration-testing) i.e. using ztest, having the name starting with `test_` etc? I am worrying that this test won't be properly recognized by the sanitycheck, but I haven't check it yet. We can discuss it furtherly in priv.

@PerMac Done.

**Comment by PerMac** [comment]
From my side it everything looks good now. I checked test execution with sanitycheck. The test are found; build correctly on: qemu_cortex_m0, nrf52840dk_nrf52840, nrf9160dk_nrf9160, nrf5340dk_nrf5340_cpuapp, nrf52dk_nrf52832; and two tests per platform ends in the results.

**Review by PerMac — APPROVED** [review]

**Comment by anangl** [comment]
@carlescufi I think this one is ready to be merged. Could you take a look?

**Comment by anangl** [comment]
Rebased on master.

**Review by tejlmand — APPROVED** [review]

**Pull request merged**